pkg/cmd/server/api/types.go

```diff
@@ -279,6 +279,21 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig MasterNetworkConfig
+
+	// JenkinsPipelineConfig holds information about the default Jenkins template
+	// used for JenkinsPipeline build strategy.
+	JenkinsPipelineConfig JenkinsPipelineConfig `json:"jenkinsPipelineConfig"`
```

Comment:
So we want to default this to enabled but make it possible to be disabled.
So we need two indicators that are distinct.  So that's usually a *bool
inside the struct.

On Tue, Apr 5, 2016 at 4:36 AM, Michal Fojtik <notifications@github.com>
wrote:

> In pkg/cmd/server/api/types.go
> <https://github.com/openshift/origin/pull/7949#discussion_r58504004>:
>
> > @@ -279,6 +279,21 @@ type MasterConfig struct {
> >
> >  	// NetworkConfig to be passed to the compiled in network plugin
> >  	NetworkConfig MasterNetworkConfig
> > +
> > +	// JenkinsPipelineConfig holds information about the default Jenkins template
> > +	// used for JenkinsPipeline build strategy.
> > +	JenkinsPipelineConfig JenkinsPipelineConfig `json:"jenkinsPipelineConfig"`
>
> @smarterclayton <https://github.com/smarterclayton> the consequence of
> making this optional is:
>
> 1) We are not able to determine the default Jenkins service name
> 2) If you don't provide the namespace/template in master config, we skip
> instantiation of Jenkins template when you create BC first time and leave
> Jenkins deployment to user
> 3) We can't provide the "default" values in master config (each admin will
> have to add the settings, which might be fine but I am afraid that will
> decrease the experience a little bit, because in order to use Jenkins
> Pipeline with automatic Jenkins deployment, you will first have to tweak
> the master configuration. It might also cause problems with all-in-one
> setup where you don't have master config, right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7949/files/e2d6ffa012a680b0d57df505a1dd9ace775dcdf3#r58504004>
>